services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_scheduler.py

```diff
@@ -248,6 +249,34 @@ async def mark_service_for_removal(
             current.dynamic_sidecar.service_removal_state.mark_to_remove(can_save)
             await update_scheduler_data_label(current)
 
+            # cancel current observation task
+            if service_name in self._service_observation_task:
+                service_task: Optional[
+                    Union[asyncio.Task, object]
+                ] = self._service_observation_task[service_name]
+                if isinstance(service_task, asyncio.Task):
+                    service_task.cancel()
+
+                    async def _wait_task(task: asyncio.Task) -> None:
+                        await task
+
+                    with suppress(asyncio.CancelledError):
+                        try:
+                            await asyncio.wait_for(_wait_task(service_task), timeout=10)
```

Comment:
This is strange ... you are "awaiting" a ``wait_for`` that awaits a task ...?? :-)

Isn't this equivalent to
```python
await asyncio.wait( service_task, timeout=10)
```
i.e. replace ``wait_for`` by ``asyncio.wait`` and do not user ``_wait_task`` but directly the task
SEE https://docs.python.org/3/library/asyncio-task.html?highlight=wait_for#waiting-primitives

Reply:
Unfortunately this is not equivalent. There is an edge case where the task that is being awaited will not yield the loop (blocking by low level primitives usually). If awaited it will just block everything. Wrapping in like this and awaiting it guarantees that after the timeout the loop is "free" again.

Reply:
that is surprising